src/commons/BalanceHandler.ts

```diff
@@ -0,0 +1,37 @@
+import { BigInt } from '@graphprotocol/graph-ts';
+import { TokenDistroBalance } from '../types/schema';
+
+export function addAllocatedTokens(
+  to: string,
+  value: BigInt,
+  tokenAddress: string,
+): void {
+  const id = tokenAddress + '-' + to;
+  let allocatedBalance = TokenDistroBalance.load(id);
+  if (!allocatedBalance) {
+    allocatedBalance = new TokenDistroBalance(id);
+  }
```

Comment:
Instead of this  code block and repeating it please implement `getUserTokenDistroBalance` like `getUserUnipoolBalance`